tests/scripts/crossgen_comparison.py

```diff
@@ -0,0 +1,372 @@
+
+def crossgen_corelib(args):
+    il_corelib_filename = args.il_corelib_filename
+    ni_corelib_filename = os.path.join(tempfile.gettempdir(), os.path.basename(il_corelib_filename))
+    crossgen_result = crossgen_assembly(args.crossgen_executable_filename, il_corelib_filename, ni_corelib_filename, [os.path.dirname(il_corelib_filename)])
+    result_filename = os.path.join(args.result_dirname, crossgen_result.assembly_name + '.json')
+    save_crossgen_result_to_json_file(crossgen_result, result_filename)
+
+def add_ni_extension(filename):
+    filename,ext = os.path.splitext(filename)
+    return filename + '.ni' + ext
+
+def crossgen_framework(args):
+    global g_Framework_Assemblies
+    platform_assemblies_paths = [args.core_root]
+    for assembly_name in g_Framework_Assemblies:
+        il_filename = os.path.join(args.core_root, assembly_name)
+        ni_filename = os.path.join(tempfile.gettempdir(), add_ni_extension(assembly_name))
+        crossgen_result = crossgen_assembly(args.crossgen_executable_filename, il_filename, ni_filename, platform_assemblies_paths)
+        result_filename = os.path.join(args.result_dirname, crossgen_result.assembly_name + '.json')
+        save_crossgen_result_to_json_file(crossgen_result, result_filename)
+
+def load_crossgen_result_from_json_file(json_filename):
+    with open(json_filename, 'rt') as json_file:
+        return json.load(json_file, cls=CrossGenResultDecoder)
+
+def load_crossgen_results_from_dir(dirname):
+    results_by_assembly_name = dict()
+    for filename in glob.glob(os.path.join(dirname, '*.json')):
+        result = load_crossgen_result_from_json_file(filename)
+        results_by_assembly_name[result.assembly_name] = result
+    return results_by_assembly_name
+
+def compare(args):
+    base_results = load_crossgen_results_from_dir(args.base_dirname)
+    diff_results = load_crossgen_results_from_dir(args.diff_dirname)
+
+    base_assemblies = set(base_results.keys())
+    diff_assemblies = set(diff_results.keys())
+
+    column_width = max(len(assembly_name) for assembly_name in base_assemblies | diff_assemblies)
+    has_mismatch_error = False
+
+    for assembly_name in sorted(base_assemblies & diff_assemblies):
+        base_result = base_results[assembly_name]
+        diff_result = diff_results[assembly_name]
+
+        if base_result.out_file_hashsum == diff_result.out_file_hashsum:
+            print('{0}  [OK]'.format(assembly_name.ljust(column_width)))
+        else:
+            print('{0}  [MISMATCH]'.format(assembly_name.ljust(column_width)))
+            has_mismatch_error = True
+
+    for assembly_name in sorted(base_assemblies - diff_assemblies):
+        print('{0}  [BASE ONLY]'.format(assembly_name.ljust(column_width)))
+        has_mismatch_error = True
+
+    for assembly_name in sorted(diff_assemblies - base_assemblies):
+        print('{0}  [DIFF ONLY]'.format(assembly_name.ljust(column_width)))
```

Comment:
Have you seen all the different success/error cases in your testing?

Reply:
Yes, when tested locally I have seen all of the cases. 

Actually, my original plan was to commit these changes as a first iteration and then continue working on getting comparison report in the script more informative.